rust/hyperlane-core/src/traits/signer.rs

```diff
@@ -0,0 +1,21 @@
+use async_trait::async_trait;
+use auto_impl::auto_impl;
+use ethers::prelude::Signature;
+use primitive_types::H256;
+use std::fmt::Debug;
+
+/// An error incurred by a signer
+#[derive(thiserror::Error, Debug)]
+#[error(transparent)]
+pub struct HyperlaneSignerError(#[from] Box<dyn std::error::Error + Send + Sync>);
+
+/// A hyperlane signer for use by the validators.
+#[async_trait]
+#[auto_impl(&, Box, Arc)]
+pub trait HyperlaneSigner: Send + Sync + Debug {
+    /// The signer's address
+    fn address(&self) -> H256;
```

Comment:
Hmm I think we should just call this `evm_address` or `ethereum_address` or `eth_address` or something and have it return an H160. Because "address" is originally an ethereum-specific thing, and we use these ethereum addresses even in the Fuel contracts when dealing with validator signatures (i.e. we recover the ethereum address from signatures, not the public key).

And then if we find this is limiting at some point in the far future, we could consider moving entirely to public keys rather than addresses

Afaict we end up converting from H256 -> H160 everywhere this is used anyways

Reply:
Yeah, that is fair, I would still want to use an H160 and not an `ethers::prelude::Address` type. Would also make it a bit more clear.